content/blog/2021-02-18-dvc-2-0-pre-release.md

```diff
@@ -0,0 +1,562 @@
+`git add/commit`, and comparing all experiments is difficult.
+
+We introduce lightweight experiments in DVC 2.0! This is the way of
+auto-tracking without any overhead from ML engineers.
+
+⚠️ Note, ML experiment is an experimental feature in the coming release. It
+means the commands might change a bit even after the release.
+
+Run an ML experiment with a new hyperparameter from `params.yaml`:
+
+```dvc
+$ dvc exp run --set-param featurize.max_features=3000
+
+Reproduced experiment(s): exp-bb55c
+Experiment results have been applied to your workspace.
+
+$ dvc exp diff
+Path         Metric    Value    Change
+scores.json  auc       0.57462  0.0072197
+
+Path         Param                   Value    Change
+params.yaml  featurize.max_features  3000     1500
+```
+
+More experiments:
+
+```dvc
+$ dvc exp run --set-param featurize.max_features=4000
+Reproduced experiment(s): exp-9bf22
+Experiment results have been applied to your workspace.
+
+$ dvc exp run --set-param featurize.max_features=5000
+Reproduced experiment(s): exp-63ee0
+Experiment results have been applied to your workspace.
+
+$ dvc exp run --set-param featurize.max_features=5000 \
+                --set-param featurize.ngrams=3
+Reproduced experiment(s): exp-80655
+Experiment results have been applied to your workspace.
+```
+
+In the examples above, hyperparamters were changed automaticaly by option
+`--set-param`. User can make this changes manualy by modifying the file. The
+same way _any code or data files can be changed_ and `dvc exp run` will capture
+the changes.
+
+See all the runs:
+
+```dvc
+$ dvc exp show --no-pager --no-timestamp \
+        --include-params featurize.max_features,featurize.ngrams
+┏━━━━━━━━━━━━━━━┳━━━━━━━━━┳━━━━━━━━━━━━━━━━━━━━━━━━┳━━━━━━━━━━━━━━━━━━┓
+┃ Experiment    ┃     auc ┃ featurize.max_features ┃ featurize.ngrams ┃
+┡━━━━━━━━━━━━━━━╇━━━━━━━━━╇━━━━━━━━━━━━━━━━━━━━━━━━╇━━━━━━━━━━━━━━━━━━┩
+│ workspace     │ 0.56359 │ 5000                   │ 3                │
+│ master        │  0.5674 │ 1500                   │ 2                │
+│ ├── exp-80655 │ 0.56359 │ 5000                   │ 3                │
+│ ├── exp-63ee0 │  0.5515 │ 5000                   │ 2                │
+│ ├── exp-9bf22 │ 0.56448 │ 4000                   │ 2                │
+│ └── exp-bb55c │ 0.57462 │ 3000                   │ 2                │
+└───────────────┴─────────┴────────────────────────┴──────────────────┘
+```
+
+Under the hood DVC uses Git to store the experiments meta-information.
+Straight-forward implementation on top of Git should include branches and
+auto-commits in the branches. This approach over-pollutes the branch namespace
+very quickly. To avoid this issue, we introduced Git custom references `exps`
+the same way as GitHub uses Git custom references `pulls` to track pull
+requests. This is an interesting technical topic that deserves a separate blog
+post. Below you can see how it works.
```

Comment:
Copy edit:
```suggestion
A straight-forward implementation would create visible branches and auto-commit
in them, but that approach would over-pollute the branch namespace very quickly.
To avoid this issue, we introduced custom Git references `exps`, the same way as
GitHub uses custom references `pulls` to track pull requests (this is an
interesting technical topic that deserves a separate blog post). Below you can
see how it works.
```